File: js_sdk/src/model/AssetAssignBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The AssetAssignBody model module.
 * @module model/AssetAssignBody
 * @version 0.1.7
 */
export class AssetAssignBody {
  /**
   * Constructs a new <code>AssetAssignBody</code>.
   * @alias module:model/AssetAssignBody
   * @class
   * @param asset_id {String} 
   * @param asset_type {String} 
   * @param bc_id {String} 
   * @param user_id {String} 
   */
  constructor(asset_id, asset_type, bc_id, user_id) {
    this.asset_id = asset_id;
    this.asset_type = asset_type;
    this.bc_id = bc_id;
    this.user_id = user_id;
  }

  /**
   * Constructs a <code>AssetAssignBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/AssetAssignBody} obj Optional instance to populate.
   * @return {module:model/AssetAssignBody} The populated <code>AssetAssignBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new AssetAssignBody();
      if (data.hasOwnProperty('advertiser_role'))
        obj.advertiser_role = ApiClient.convertToType(data['advertiser_role'], 'String');
      if (data.hasOwnProperty('asset_id'))
        obj.asset_id = ApiClient.convertToType(data['asset_id'], 'String');
      if (data.hasOwnProperty('asset_type'))
        obj.asset_type = ApiClient.convertToType(data['asset_type'], 'String');
      if (data.hasOwnProperty('bc_id'))
        obj.bc_id = ApiClient.convertToType(data['bc_id'], 'String');
      if (data.hasOwnProperty('catalog_role'))
        obj.catalog_role = ApiClient.convertToType(data['catalog_role'], 'String');
      if (data.hasOwnProperty('form_library_role'))
        obj.form_library_role = ApiClient.convertToType(data['form_library_role'], 'String');
      if (data.hasOwnProperty('store_role'))
        obj.store_role = ApiClient.convertToType(data['store_role'], 'String');
      if (data.hasOwnProperty('tt_account_roles'))
        obj.tt_account_roles = ApiClient.convertToType(data['tt_account_roles'], ['String']);
      if (data.hasOwnProperty('user_id'))
        obj.user_id = ApiClient.convertToType(data['user_id'], 'String');
    }
    return obj;
  }
}

/**
 * @member {String} advertiser_role
 */
AssetAssignBody.prototype.advertiser_role = undefined;

/**
 * @member {String} asset_id
 */
AssetAssignBody.prototype.asset_id = undefined;

/**
 * @member {String} asset_type
 */
AssetAssignBody.prototype.asset_type = undefined;

/**
 * @member {String} bc_id
 */
AssetAssignBody.prototype.bc_id = undefined;

/**
 * @member {String} catalog_role
 */
AssetAssignBody.prototype.catalog_role = undefined;

/**
 * @member {String} form_library_role
 */
AssetAssignBody.prototype.form_library_role = undefined;

/**
 * @member {String} store_role
 */
AssetAssignBody.prototype.store_role = undefined;

/**
 * @member {Array.<String>} tt_account_roles
 */
AssetAssignBody.prototype.tt_account_roles = undefined;

/**
 * @member {String} user_id
 */
AssetAssignBody.prototype.user_id = undefined;

File: java_sdk/src/main/java/business_api_client/AssetAssignBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * AssetAssignBody
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-07-03T14:36:18.961395-07:00[America/Los_Angeles]")
public class AssetAssignBody {
  @JsonProperty("advertiser_role")
  private String advertiserRole = null;

  @JsonProperty("asset_id")
  private String assetId = null;

  @JsonProperty("asset_type")
  private String assetType = null;

  @JsonProperty("bc_id")
  private String bcId = null;

  @JsonProperty("catalog_role")
  private String catalogRole = null;

  @JsonProperty("form_library_role")
  private String formLibraryRole = null;

  @JsonProperty("store_role")
  private String storeRole = null;

  @JsonProperty("tt_account_roles")
  private List<String> ttAccountRoles = null;

  @JsonProperty("user_id")
  private String userId = null;

  public AssetAssignBody advertiserRole(String advertiserRole) {
    this.advertiserRole = advertiserRole;
    return this;
  }

   /**
   * Get advertiserRole
   * @return advertiserRole
  **/
  @Schema(description = "")
  public String getAdvertiserRole() {
    return advertiserRole;
  }

  public void setAdvertiserRole(String advertiserRole) {
    this.advertiserRole = advertiserRole;
  }

  public AssetAssignBody assetId(String assetId) {
    this.assetId = assetId;
    return this;
  }

   /**
   * Get assetId
   * @return assetId
  **/
  @Schema(required = true, description = "")
  public String getAssetId() {
    return assetId;
  }

  public void setAssetId(String assetId) {
    this.assetId = assetId;
  }

  public AssetAssignBody assetType(String assetType) {
    this.assetType = assetType;
    return this;
  }

   /**
   * Get assetType
   * @return assetType
  **/
  @Schema(required = true, description = "")
  public String getAssetType() {
    return assetType;
  }

  public void setAssetType(String assetType) {
    this.assetType = assetType;
  }

  public AssetAssignBody bcId(String bcId) {
    this.bcId = bcId;
    return this;
  }

   /**
   * Get bcId
   * @return bcId
  **/
  @Schema(required = true, description = "")
  public String getBcId() {
    return bcId;
  }

  public void setBcId(String bcId) {
    this.bcId = bcId;
  }

  public AssetAssignBody catalogRole(String catalogRole) {
    this.catalogRole = catalogRole;
    return this;
  }

   /**
   * Get catalogRole
   * @return catalogRole
  **/
  @Schema(description = "")
  public String getCatalogRole() {
    return catalogRole;
  }

  public void setCatalogRole(String catalogRole) {
    this.catalogRole = catalogRole;
  }

  public AssetAssignBody formLibraryRole(String formLibraryRole) {
    this.formLibraryRole = formLibraryRole;
    return this;
  }

   /**
   * Get formLibraryRole
   * @return formLibraryRole
  **/
  @Schema(description = "")
  public String getFormLibraryRole() {
    return formLibraryRole;
  }

  public void setFormLibraryRole(String formLibraryRole) {
    this.formLibraryRole = formLibraryRole;
  }

  public AssetAssignBody storeRole(String storeRole) {
    this.storeRole = storeRole;
    return this;
  }

   /**
   * Get storeRole
   * @return storeRole
  **/
  @Schema(description = "")
  public String getStoreRole() {
    return storeRole;
  }

  public void setStoreRole(String storeRole) {
    this.storeRole = storeRole;
  }

  public AssetAssignBody ttAccountRoles(List<String> ttAccountRoles) {
    this.ttAccountRoles = ttAccountRoles;
    return this;
  }

  public AssetAssignBody addTtAccountRolesItem(String ttAccountRolesItem) {
    if (this.ttAccountRoles == null) {
      this.ttAccountRoles = new ArrayList<String>();
    }
    this.ttAccountRoles.add(ttAccountRolesItem);
    return this;
  }

   /**
   * Get ttAccountRoles
   * @return ttAccountRoles
  **/
  @Schema(description = "")
  public List<String> getTtAccountRoles() {
    return ttAccountRoles;
  }

  public void setTtAccountRoles(List<String> ttAccountRoles) {
    this.ttAccountRoles = ttAccountRoles;
  }

  public AssetAssignBody userId(String userId) {
    this.userId = userId;
    return this;
  }

   /**
   * Get userId
   * @return userId
  **/
  @Schema(required = true, description = "")
  public String getUserId() {
    return userId;
  }

  public void setUserId(String userId) {
    this.userId = userId;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AssetAssignBody assetAssignBody = (AssetAssignBody) o;
    return Objects.equals(this.advertiserRole, assetAssignBody.advertiserRole) &&
        Objects.equals(this.assetId, assetAssignBody.assetId) &&
        Objects.equals(this.assetType, assetAssignBody.assetType) &&
        Objects.equals(this.bcId, assetAssignBody.bcId) &&
        Objects.equals(this.catalogRole, assetAssignBody.catalogRole) &&
        Objects.equals(this.formLibraryRole, assetAssignBody.formLibraryRole) &&
        Objects.equals(this.storeRole, assetAssignBody.storeRole) &&
        Objects.equals(this.ttAccountRoles, assetAssignBody.ttAccountRoles) &&
        Objects.equals(this.userId, assetAssignBody.userId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(advertiserRole, assetId, assetType, bcId, catalogRole, formLibraryRole, storeRole, ttAccountRoles, userId);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AssetAssignBody {\n");
    
    sb.append("    advertiserRole: ").append(toIndentedString(advertiserRole)).append("\n");
    sb.append("    assetId: ").append(toIndentedString(assetId)).append("\n");
    sb.append("    assetType: ").append(toIndentedString(assetType)).append("\n");
    sb.append("    bcId: ").append(toIndentedString(bcId)).append("\n");
    sb.append("    catalogRole: ").append(toIndentedString(catalogRole)).append("\n");
    sb.append("    formLibraryRole: ").append(toIndentedString(formLibraryRole)).append("\n");
    sb.append("    storeRole: ").append(toIndentedString(storeRole)).append("\n");
    sb.append("    ttAccountRoles: ").append(toIndentedString(ttAccountRoles)).append("\n");
    sb.append("    userId: ").append(toIndentedString(userId)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
